fairmq/sdk/Topology.h

```diff
@@ -250,11 +251,18 @@ class BasicTopology : public AsioBase<Executor, Allocator>
     BasicTopology(BasicTopology&&) = default;
     BasicTopology& operator=(BasicTopology&&) = default;
 
-    ~BasicTopology()
+    void UnsubscribeFromStateChanges()
     {
-        
+        using namespace fair::mq::sdk::cmd;
 
+        fDDSSession.SendCommand(Cmds(make<UnsubscribeFromStateChange>()).Serialize());
```

Comment:
Thinking about this: Currently, there is a n-to-1 relationship between `sdk::Topology` objects and `sdk::DDSSession::Impl` objects. In the current implementation (also already before this commit) a second instantiation of a `sdk::Topology` for the same DDS session will result in two state change subscriptions from the same DDS channel ID which then will be folded by DDS to one really, which is probably fine for subscribe, but the first unsubscribe will also unsubscribe all other `sdk::Topology`s. Also, not sure, what happens, if the last custom command callback overrides previous ones? So, we should probably improve here at some point and add a (de)multiplexer to `sdk::DDSSession`.

Reply:
(Out of scope of this PR)